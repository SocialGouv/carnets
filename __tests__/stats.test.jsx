import { vi, expect, test } from "vitest";
import { render } from "@testing-library/react";
import Page from "../src/app/stats/page";

vi.mock("@/actions/get-stats", async (importOriginal) => {
  const actual = await importOriginal();

  return {
    ...actual,
    default: () => ({
      posts_stat: [
        { count: 5, week: 6, year: 2020 },
        { count: 2, week: 7, year: 2020 },
        { count: 1, week: 8, year: 2020 },
        { count: 4, week: 9, year: 2020 },
        { count: 4, week: 10, year: 2020 },
        { count: 7, week: 11, year: 2020 },
        { count: 3, week: 12, year: 2020 },
        { count: 4, week: 13, year: 2020 },
        { count: 8, week: 14, year: 2020 },
        { count: 5, week: 15, year: 2020 },
        { count: 7, week: 16, year: 2020 },
        { count: 7, week: 17, year: 2020 },
        { count: 8, week: 18, year: 2020 },
        { count: 7, week: 19, year: 2020 },
        { count: 9, week: 20, year: 2020 },
        { count: 6, week: 21, year: 2020 },
        { count: 6, week: 22, year: 2020 },
        { count: 6, week: 23, year: 2020 },
        { count: 7, week: 24, year: 2020 },
        { count: 7, week: 25, year: 2020 },
        { count: 7, week: 26, year: 2020 },
        { count: 6, week: 27, year: 2020 },
        { count: 8, week: 28, year: 2020 },
        { count: 3, week: 30, year: 2020 },
        { count: 5, week: 31, year: 2020 },
        { count: 8, week: 32, year: 2020 },
        { count: 5, week: 33, year: 2020 },
        { count: 6, week: 34, year: 2020 },
        { count: 7, week: 35, year: 2020 },
        { count: 4, week: 36, year: 2020 },
        { count: 5, week: 37, year: 2020 },
        { count: 7, week: 38, year: 2020 },
        { count: 7, week: 39, year: 2020 },
        { count: 7, week: 40, year: 2020 },
        { count: 8, week: 41, year: 2020 },
        { count: 9, week: 42, year: 2020 },
        { count: 7, week: 43, year: 2020 },
        { count: 8, week: 44, year: 2020 },
        { count: 8, week: 45, year: 2020 },
        { count: 10, week: 46, year: 2020 },
        { count: 10, week: 47, year: 2020 },
        { count: 7, week: 48, year: 2020 },
        { count: 9, week: 49, year: 2020 },
        { count: 8, week: 50, year: 2020 },
        { count: 5, week: 51, year: 2020 },
        { count: 4, week: 52, year: 2020 },
        { count: 3, week: 53, year: 2020 },
        { count: 7, week: 1, year: 2021 },
        { count: 5, week: 2, year: 2021 },
        { count: 6, week: 3, year: 2021 },
        { count: 3, week: 4, year: 2021 },
        { count: 5, week: 5, year: 2021 },
        { count: 4, week: 6, year: 2021 },
        { count: 6, week: 7, year: 2021 },
        { count: 7, week: 8, year: 2021 },
        { count: 10, week: 9, year: 2021 },
        { count: 7, week: 10, year: 2021 },
        { count: 5, week: 11, year: 2021 },
        { count: 5, week: 12, year: 2021 },
        { count: 7, week: 13, year: 2021 },
        { count: 9, week: 14, year: 2021 },
        { count: 7, week: 15, year: 2021 },
        { count: 8, week: 16, year: 2021 },
        { count: 9, week: 17, year: 2021 },
        { count: 11, week: 18, year: 2021 },
      ],
      teams_status: [
        {
          mood: "average",
          kpis_count: 4,
          last_post: "2023-12-14T10:26:11.394668+00:00",
          team_slug: "archifiltre",
        },
        {
          mood: "good",
          kpis_count: 3,
          last_post: "2023-03-07T08:48:24.382673+00:00",
          team_slug: "bpco",
        },
        {
          mood: "average",
          kpis_count: 2,
          last_post: "2023-12-14T09:37:06.725605+00:00",
          team_slug: "bpco-mieux",
        },
        {
          mood: "good",
          kpis_count: 3,
          last_post: "2021-12-07T10:59:53.763344+00:00",
          team_slug: "carnet-de-bord",
        },
        {
          mood: "average",
          kpis_count: 1,
          last_post: "2023-06-06T07:41:35.199012+00:00",
          team_slug: "causes-medicales-de-deces",
        },
        {
          mood: "good",
          kpis_count: 3,
          last_post: "2023-12-14T08:29:20.133928+00:00",
          team_slug: "code-du-travail",
        },
        {
          mood: "average",
          kpis_count: 1,
          last_post: "2021-01-05T10:33:03.544189+00:00",
          team_slug: "covid-19",
        },
        {
          mood: "average",
          kpis_count: 1,
          last_post: "2022-03-29T09:34:22.430957+00:00",
          team_slug: "dashlord",
        },
        {
          mood: "average",
          kpis_count: 1,
          last_post: "2020-12-15T10:40:01.458244+00:00",
          team_slug: "data-dgt",
        },
        {
          mood: "average",
          kpis_count: 1,
          last_post: "2021-10-19T08:21:11.02241+00:00",
          team_slug: "didoc",
        },
        {
          mood: "average",
          kpis_count: 2,
          last_post: "2021-05-10T14:09:59.932222+00:00",
          team_slug: "diversite",
        },
      ],
    }),
  };
});

test("Page", async () => {
  const { container } = render(await Page());
  expect(container).toMatchSnapshot();
});
