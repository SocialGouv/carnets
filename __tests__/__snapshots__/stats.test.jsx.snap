// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page 1`] = `
<div>
  <main>
    <section>
      <h2
        class="fr-h2"
      >
        Nombre de publications par semaine
      </h2>
      <div
        class="h-96 relative mb-12"
      >
        <canvas
          height="150"
          role="img"
          width="300"
        />
      </div>
    </section>
    <section>
      <h2
        class="fr-h2"
      >
        Statuts des produits
      </h2>
      <div
        class="fr-table fr-table--layout-fixed fr-table--no-caption"
        id="fr-table-:r0:"
      >
        <table>
          <thead>
            <tr>
              <th
                scope="col"
              >
                Produit
              </th>
              <th
                scope="col"
              >
                État d'esprit
              </th>
              <th
                scope="col"
              >
                Dernière publication
              </th>
              <th
                scope="col"
              >
                Nombre de KPIs
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                archifiltre
              </td>
              <td>
                average
              </td>
              <td>
                14-12-2023
              </td>
              <td>
                4
              </td>
            </tr>
            <tr>
              <td>
                bpco
              </td>
              <td>
                good
              </td>
              <td>
                07-03-2023
              </td>
              <td>
                3
              </td>
            </tr>
            <tr>
              <td>
                bpco-mieux
              </td>
              <td>
                average
              </td>
              <td>
                14-12-2023
              </td>
              <td>
                2
              </td>
            </tr>
            <tr>
              <td>
                carnet-de-bord
              </td>
              <td>
                good
              </td>
              <td>
                07-12-2021
              </td>
              <td>
                3
              </td>
            </tr>
            <tr>
              <td>
                causes-medicales-de-deces
              </td>
              <td>
                average
              </td>
              <td>
                06-06-2023
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                code-du-travail
              </td>
              <td>
                good
              </td>
              <td>
                14-12-2023
              </td>
              <td>
                3
              </td>
            </tr>
            <tr>
              <td>
                covid-19
              </td>
              <td>
                average
              </td>
              <td>
                05-01-2021
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                dashlord
              </td>
              <td>
                average
              </td>
              <td>
                29-03-2022
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                data-dgt
              </td>
              <td>
                average
              </td>
              <td>
                15-12-2020
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                didoc
              </td>
              <td>
                average
              </td>
              <td>
                19-10-2021
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                diversite
              </td>
              <td>
                average
              </td>
              <td>
                10-05-2021
              </td>
              <td>
                2
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </section>
  </main>
</div>
`;
